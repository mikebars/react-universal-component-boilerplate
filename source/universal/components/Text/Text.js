/* @flow */

import styled from 'styled-components'

export const Text = styled.span`
  font-size: ${props => props.fontSize};
`

export default Text
