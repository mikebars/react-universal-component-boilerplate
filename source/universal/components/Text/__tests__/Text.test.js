/* @flow */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount, shallow, render } from 'enzyme'

import { Text } from 'universal/components/Text/Text'

test('Text - snapshot test', () => {
  const props = {}

  const element = <Text {...props} />

  const tree = renderer.create(element).toJSON()

  expect(tree).toMatchSnapshot()
})

test('Text - enzyme shallow render', () => {
  const props = {}

  const element = <Text {...props} />

  const wrapper = shallow(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('Text - enzyme full render', () => {
  const props = {}

  const element = <Text {...props} />

  const wrapper = mount(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('Text - enzyme static render', () => {
  const props = {}

  const element = <Text {...props} />

  const wrapper = render(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})
