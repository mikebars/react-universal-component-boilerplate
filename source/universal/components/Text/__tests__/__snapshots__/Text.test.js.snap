// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text - snapshot test 1`] = `
<span
  className="Text-ia49n9-0 geatmR"
/>
`;
